src/main/java/br/com/caelum/mamute/user/domain/service/UserServiceImpl.java

```diff
@@ -32,19 +32,22 @@
     public UserEntity signup(final UserEntity user) {
         this.validate(user);
         final UserEntity userPersisted = this.userRepository.save(user);
-        final LoginMethod loginMethodCreated = this.loginMethodRepository
-                .save(userPersisted.getLoginMethods().get(FIRST_INDEX));
-
+        this.loginMethodRepository.save(userPersisted.getLoginMethods().get(FIRST_INDEX));
         return userPersisted;
     }
 
     @Override
     @LogException(exceptions = { Exception.class })
-    public Page<UserEntity> findUserByFilter(SearchCriteria criteria, Pageable pageable) {
-
-        final PageRequest page = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.Direction.ASC);
+    public Page<UserEntity> findUserByFilter(final UserEntity filter, final Pageable pageable) {
+        final PageRequest page = this.createPageRequest(pageable);
+        return this.userRepository.findAll(this.userRepository.toSpecification(filter), page);

```

Comment:
Quando fazemos `this.userRepository.toSpecification(filter)` não estamos passando um detalhe de implementação de uma camada inferior para a camada de cima? 
Se abstrairmos isso no repository, e o service apenas informar o `filter` e o `page`?

Reply:
mano, concordo, ajustado.